homeassistant/components/rflink/binary_sensor.py

```diff
@@ -1,26 +1,38 @@
 """Support for Rflink binary sensors."""
+import logging
+
 import voluptuous as vol
 
 from homeassistant.components.binary_sensor import (
     DEVICE_CLASSES_SCHEMA,
     PLATFORM_SCHEMA,
     BinarySensorEntity,
 )
-from homeassistant.const import CONF_DEVICE_CLASS, CONF_FORCE_UPDATE, CONF_NAME
+from homeassistant.const import (
+    CONF_DEVICE_CLASS,
+    CONF_FORCE_UPDATE,
+    CONF_NAME,
+    CONF_TYPE,
+)
 import homeassistant.helpers.config_validation as cv
 import homeassistant.helpers.event as evt
 
 from . import CONF_ALIASES, CONF_DEVICES, RflinkDevice
 
+_LOGGER = logging.getLogger(__name__)
+
 CONF_OFF_DELAY = "off_delay"
 DEFAULT_FORCE_UPDATE = False
+TYPE_STANDARD = "standard"
+TYPE_INVERTED = "inverted"
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_DEVICES, default={}): {
             cv.string: vol.Schema(
                 {
                     vol.Optional(CONF_NAME): cv.string,
+                    vol.Optional(CONF_TYPE): vol.Any(TYPE_STANDARD, TYPE_INVERTED),
```

Comment:
What other types can we expect in the future?
(As in; why isn't this a simple `inverted` boolean parameter?)

Reply:
I see your point.
I'm not sure if in the future new types will be created, but this implementation follows other platform implementations like:
* light: https://www.home-assistant.io/integrations/light.rflink/#type
* cover: https://www.home-assistant.io/integrations/cover.rflink/#type

Although it is true that it could be replaceed by a boolean, I think keeping a uniform criterion on the platform helps to understand the use of the attribute.

Reply:
I have reviewed the use of the `type` attribute in the rest of the platforms and although the` inverted` type is also used in the `cover` platform, making use of an attribute instead of a type would have advantages in the implementation by decoupling the cover's behavior (defined by the type) from the action commands (`up`/`down`).
I don't know if new types of binary sensors could emerge in the future, but they would also benefit from a decoupled approach .

Therefore, the option to replace the type with an attribute does not seem bad to me.